src/fsharp/FSharp.Core/mailbox.fs

```diff
@@ -325,6 +325,11 @@ namespace Microsoft.FSharp.Control
     type AsyncReplyChannel<'Reply>(replyf : 'Reply -> unit) =
         member x.Reply value = replyf value
 
+    module private AsyncStarter =
```

Comment:
I think remove this module, and inline each of its elements.  They don't make code any clearer, thanks

Reply:
Done

Reply:
Yes, we need more testing, please add something that checks that a StartImmediate really happened